### PR TITLE
update readme

### PR DESCRIPTION
**Pull request by sfc-gh-aling**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,31 +1,37 @@
 # Introduction
 
-snowflake-vcrpy is designed to enhance local development experiences for applications
-built on top of snowflake-connector-python. Libraries like snowpark-python or snowflake-sqlalchemy which depends
-on the connector for http communication can also take advantage of the tool.
+snowflake-vcrpy is a library designed to allow you to record and replay HTTP interactions initiated by snowflake-connector-python for testing purposes. It is a pytest plugin built upon [VCR.py][vcrpy].
```
Comment (sfc-gh-sfan) [inline README.md]: "designed to allow" -> "that allows" ?

File: README.md [hunk 2]
```diff
@@ -1,31 +1,37 @@
...
+ Libraries like snowpark-python or snowflake-sqlalchemy depending
+on the connector for HTTP interactions can also take advantage of the library.
```
Comment (sfc-gh-sfan) [inline README.md]: Feels like line 3 is much longer than line 4-5 combined

File: README.md [hunk 3]
```diff
@@ -1,31 +1,37 @@
...
 
 snowflake-vcrpy provides the following features:
 
-1. Record and replay http based tests for faster local development
+1. Pytest fixuture to enable record and replay http based tests
```
Comment (sfc-gh-sfan) [inline README.md]: typo: `fixuture` -> `fixture`

File: README.md [hunk 4]
```diff
@@ -1,31 +1,37 @@
...
+2. Pytest options to select the tests run in record and replay mode
 
 
-# Installation from source code
+# Installation
+
+snowflake-vcrpy can be installed from soure code:
```
Comment (sfc-gh-sfan) [inline README.md]: typo: `soure` -> `source`

File: README.md [hunk 5]
```diff
@@ -1,31 +1,37 @@
...
-$pip install .
+$ git clone git@github.com:Snowflake-Labs/snowflake-vcrpy.git
+$ cd snowflake-vcrpy
+$ pip install .
 ```
 
 # Quick Start
 
-## Annotate a test to run with record and replay
+## Annotate a test with pytest marker to run in record and replay
```
Comment (sfc-gh-sfan) [inline README.md]: `record and replay` -> `record-and-replay mode`?

File: README.md [hunk 6]
```diff
@@ -35,18 +41,20 @@ snowflake-vcrpy is a pytest plugin and by default it will identify tests annotat
...
-$pytest test_method
+$pytest <test_file.py>
 ```
 
-With the first time running the test, a recording file containing http request/response with
-the name being `<test_class_name>,<test_method_name>.yaml` will be
-generated under the directory `<your_test_folder>/cassettes`.
+## Rationale
+
+The utilization of [VCR.py][vcrpy] is incorporated into snowflake-vcrpy to facilitate the process of recording and replaying.
```
Comment (sfc-gh-sfan) [inline README.md]: How about `snowflake-vcrpy incorporates [VCR.py][vcrpy] to...`

File: README.md [hunk 7]
```diff
@@ -35,18 +41,20 @@ snowflake-vcrpy is a pytest plugin and by default it will identify tests annotat
...
 
-Once the recording is generated, the following running of the test will run against the local recordings.
+Upon the initial execution of the test, a recording file, consisting of HTTP requests and responses, will be generated and saved under the directory `<your_test_folder>/cassettes` with the naming convention of `<test_class_name>.<test_method_name>.yaml`.
 
-If any change is made upon the test itself on purpose which changes the body of the request/response, you could delete
-the specific recording file under `<your_test_folder>/cassettes` so that the next running will regenerate the recording.
+VCR.py will retrieve the serialized requests and responses stored in the cassette file, and if it encounters any HTTP requests from the original test run, it will intercept them and return the relevant responses that were previously recorded.
```
Comment (sfc-gh-sfan) [inline README.md]: `relevant responses` -> `corresponding responses`?

File: README.md (hunk 2 above)
Comment (sfc-gh-aling) [reply README.md]: they will be rendered the same, but yeah it's good to split line3 for better maintenance.

**Review by sfc-gh-sfan — APPROVED** [review]

File: README.md [hunk 8]
```diff
@@ -1,54 +1,63 @@
...
-on the connector for http communication can also take advantage of the tool.
+snowflake-vcrpy is a library that allows you to record and replay HTTP interactions initiated by
+snowflake-connector-python for testing purposes. It is a pytest plugin built upon [VCR.py][vcrpy].
+Libraries like snowpark-python and snowflake-sqlalchemy depending
+on the connector for HTTP interactions can also leverage the library.
 
 snowflake-vcrpy provides the following features:
 
-1. Record and replay http based tests for faster local development
+1. pytest fixtures to enable record and replay http based tests
```
Comment (sfc-gh-mabrennan) [inline README.md]: Maybe replace "http based tests" with "HTTP-based tests".

**Review by sfc-gh-mabrennan — APPROVED** [review]
lgtm

**Pull request merged**